src/react/hooks/use-typing.ts

```diff
@@ -0,0 +1,100 @@
+import { Typing, TypingEvent, TypingListener } from '@ably/chat';
+import { useCallback, useEffect, useState } from 'react';
+
+import { ChatStatusResponse } from '../types/chat-status-response.js';
+import { Listenable } from '../types/listenable.js';
+import { StatusParams } from '../types/status-params.js';
+import { useChatConnection } from './use-chat-connection.js';
+import { useRoom } from './use-room.js';
+
+/**
+ * The options for the {@link useTyping} hook.
+ */
+export interface TypingParams extends StatusParams, Listenable<TypingListener> {
+  /**
+   * A listener that will be called whenever a typing event is sent to the room.
+   */
+  listener?: TypingListener;
+}
+
+export interface UseTypingResponse extends ChatStatusResponse {
+  /**
+   * A shortcut to the {@link Typing.start} method.
+   */
+  readonly start: Typing['start'];
+
+  /**
+   * A shortcut to the {@link Typing.stop} method.
+   */
+  readonly stop: Typing['stop'];
+
+  /**
+   * The set of client IDs that are currently typing in the room.
+   */
+  readonly currentlyTyping: TypingEvent['currentlyTyping'];
+
+  /**
+   * Provides access to the underlying {@link Typing} instance of the room.
+   */
+  readonly typingIndicators: Typing;
+}
+
+/**
+ * A hook that provides access to the {@link Typing} instance in the room.
+ * It will use the instance belonging to the room in the nearest {@link RoomProvider} in the component tree.
+ *
+ * @param params - Allows the registering of optional callbacks.
+ * @returns UseTypingResponse - An object containing the {@link Typing} instance and methods to interact with it.
+ */
+export const useTyping = (params?: TypingParams): UseTypingResponse => {
+  const { currentStatus: connectionStatus, error: connectionError } = useChatConnection({
+    onStatusChange: params?.onConnectionStatusChange,
+  });
+  const { room, roomError, roomStatus } = useRoom({
+    onStatusChange: params?.onRoomStatusChange,
+  });
+
+  const [currentlyTyping, setCurrentlyTyping] = useState<Set<string>>(new Set());
+
+  useEffect(() => {
+    const { unsubscribe } = room.typing.subscribe((event) => {
```

Comment:
Should this call `get()` before it subscribes to make sure the set is up-to-date on page/room load even if there are no typing events coming?

Reply:
Yeh that's a good point, I'll add this as part of the mount process and do the same for presence :)